src/dbnode/persist/fs/seek_manager_test.go

```diff
@@ -680,3 +681,55 @@ func TestSeekerManagerCacheShardIndicesSkipNotFound(t *testing.T) {
 
 	require.NoError(t, m.Close())
 }
+
+func TestSeekerManagerDoNotOpenSeekersForOutOfRetentionBlocks(t *testing.T) {
+	defer leaktest.CheckTimeout(t, 1*time.Minute)()
+	var (
+		ctrl        = xtest.NewController(t)
+		shards      = []uint32{0}
+		metadata    = testNs1Metadata(t)
+		rOpts       = metadata.Options().RetentionOptions()
+		blockSize   = rOpts.BlockSize()
+		signal      = make(chan struct{})
+		openSeekers = make(map[xtime.UnixNano]struct{})
+		now         = time.Now()
```

Comment:
Should we maybe align this on block start in order to cover `require.Contains(t, openSeekers, earliestBlockStart)` more precisely?